src/Validator.php

```diff
@@ -44,36 +39,21 @@ class Validator
      * $conditions - array of conditions
      * $then_rules - array in $this->rules format which will be used if conditions are met
      * $else_rules - array in $this->rules format which will be used if conditions are not met
-     *
-     * @var array
      */
-    public $if_rules = [];
+    public array $if_rules = [];
 
-    /**
-     * Initialization.
-     */
     public function __construct(Model $model)
     {
-        $this->model = $model;
-
-        if (!isset($model->validator)) {
-            $model->validator = $this;
-        }
-
-        $model->onHook($model::HOOK_VALIDATE, \Closure::fromCallable([$this, 'validate']));
+        $model->onHook(Model::HOOK_VALIDATE, \Closure::fromCallable([$this, 'validate']));
     }
 
     /**
      * Set one rule.
-     *
-     * @param array|string|callable $rules
-     *
-     * @return $this
      */
-    public function rule(string $field, $rules)
+    public function rule(string $field, array $rules): self
```

Comment:
`$this` return val should be kept

Reply:
in the @return docblock for referencing static, no?

Reply:
? I seen `$this` here, which is much stronger type than `self` or even `static`.

Reply:
now i add `@return $this` in docblock and i leave `:self` in the signature, it's ok, if not show me what you suggest

Reply:
👍, `$this` is needed to help phpstan to understand/check it is the same instance, also to help the developer to know it is return for method chaining